app/controllers/api/connections_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Api::ConnectionsController < ApplicationController
+    include AuthenticationConcern
+
+    def request_match
+
+        if @current_user
+
+            request = MatchRequest.new
+            request.student_id = @current_user.id
+
+            if SkillClass.find_by(id: params[:id])
+                #check if class exists
+                request.class_id = params[:id]
+
+                if MatchRequest.find_by(student_id: @current_user.id, class_id: params[:id])
+                    #check if request already exists
+                    error = {"error": "Match request already exists"}
+                    render(json: error, status: 400)
+                    return
+                end
+
+                if request.save
```

Comment:
This looks ok, but I'd probably remove the `return` and instead merge the `end` and `if` into an `elsif` (so, after the line with the `render ...`, you'd have a line with `elsif request.save`)